Brinstar Reserve canSpeedKeep

blkerby opened the pull request:
I don't know exactly how this is done (haven't practiced speed keeps) but I remember seeing it on a stream before. @osse101 would you mind checking?

osse101 commented:
https://youtu.be/TpRe3x84Sts

-canMockBall
-A way to describe entering the room while falling and not morphed.  It looks possible to run into the room with very slow run speed and jump in room too.  I wanted to make a tech for `canLowRunSpeedShortcharge` already for other strats.
-I'd call jumping into the bomb blocks a tricky jump.  Holding onto blue in the air would be part of speedkeep.
-An alternate strat using springball instead of speedkeep, which is a bit easier.

blkerby commented:
Thanks, I added canMockball, canTrickyJump, a SpringBall option, and changed it to use adjacentRunway + canShineCharge (instead of canComeInCharged) to encode the ability to jump through the door from the previous room while blue with low enough speed. This isn't perfect since other runway lengths could also work (assuming a different level of short-charging ability), but this should handle most cases, and we can always refine this later once we have a proper way to express jumping out of and into a room with blue speed.

I realized that canComeInCharged wouldn't work correctly here since it could be (wrongly) satisfied by combining with the in-room runway or by charging a spark and entering the room without blue.

osse101 commented:
👍
